concent_api/verifier/utils.py

```diff
@@ -96,3 +97,46 @@ def delete_file(file_path):
             os.unlink(file_path)
     except OSError as exception:
         logger.warning(f'File with path {file_path} was not deleted, exception: {exception}')
+
+
+def upload_file_to_storage_cluster(file_content, file_path, upload_token):
+    dumped_upload_token = dump(upload_token, None, settings.CONCENT_PUBLIC_KEY)
+    b64_encoded_token = b64encode(dumped_upload_token).decode()
+    headers = {
+        'Authorization': 'Golem ' + b64_encoded_token,
+        'Concent-Auth': b64encode(
+            dump(
+                message.concents.ClientAuthorization(
+                    client_public_key=settings.CONCENT_PUBLIC_KEY,
+                ),
+                settings.CONCENT_PRIVATE_KEY,
+                settings.CONCENT_PUBLIC_KEY
+            ),
+        ).decode(),
+        'Concent-upload-path': file_path,
+        'Content-Type': 'application/octet-stream'
+    }
+    return requests.post(
+        "{}upload/".format(settings.STORAGE_CLUSTER_ADDRESS),
+        headers=headers,
+        data=file_content,
+        verify=False
+    )
+
+
+def generate_blender_output_file_name(scene_file):
+    return f'{settings.VERIFIER_STORAGE_PATH}/{scene_file}_out'
+
+
+def generate_upload_file_name(subtask_id):
+    return f'blender/verifier-output/{subtask_id}/{subtask_id}'
+
+
+def get_file_size_and_checksum(file_path):
+    with open(file_path, 'r') as file:
```

Comment:
if `file` fails to be read, `file_content` will be udefined... Maybe this function should do something more explicit in case that reading the file fails?

Reply:
Right, how about wrapping the usage of it in try/except OsError ?

Reply:
:+1: done

Reply:
WTF. Why are we computing the size and checksum here? You're supposed to get this information in `blender_verification_order`.

Reply:
Oh, this is a file we generate ourselves. In that case it's OK.

Reply:
> Maybe this function should do something more explicit in case that reading the file fails?

In normal circumstances I would say it should just fail. For example if you can't read it due to permissions, something must be broken in container configuration. The whole verification should fail.

But since this step is not critical (it's only for auditing), I'd say that cathing the exception, silencing it and sending a crash report to Sentry would be fine too.

Reply:
This code was moved directly to `task.py` and is now handled differently.